pkg/service/admission/externalipranger/externalip_admission.go

```diff
@@ -1,25 +1,66 @@
-package admission
+package externalipranger
 
 import (
+	"fmt"
 	"io"
 	"net"
+	"reflect"
 	"strings"
 
 	apierrs "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/util/validation/field"
 	admission "k8s.io/apiserver/pkg/admission"
 	kapi "k8s.io/kubernetes/pkg/apis/core"
+
+	"github.com/golang/glog"
+	configlatest "github.com/openshift/origin/pkg/cmd/server/apis/config/latest"
+	"github.com/openshift/origin/pkg/service/admission/apis/externalipranger"
 )
 
 const ExternalIPPluginName = "ExternalIPRanger"
 
 func RegisterExternalIP(plugins *admission.Plugins) {
 	plugins.Register("ExternalIPRanger",
 		func(config io.Reader) (admission.Interface, error) {
-			return NewExternalIPRanger(nil, nil, false), nil
+			pluginConfig, err := readConfig(config)
+			if err != nil {
+				return nil, err
+			}
+			if pluginConfig == nil {
+				glog.Infof("Admission plugin %q is not configured so it will be disabled.", ExternalIPPluginName)
+				return nil, nil
+			}
+
+			// this needs to be moved upstream to be part of core config
+			reject, admit, err := ParseRejectAdmitCIDRRules(pluginConfig.ExternalIPNetworkCIDRs)
+			if err != nil {
+				// should have been caught with validation
+				return nil, err
+			}
+
+			return NewExternalIPRanger(reject, admit, pluginConfig.AllowIngressIP), nil
 		})
 }
 
+func readConfig(reader io.Reader) (*externalipranger.ExternalIPRangerAdmissionConfig, error) {
+	if reader == nil || reflect.ValueOf(reader).IsNil() {
```

Comment:
oh, wow. we really still need the second case?

Reply:
such a nil is a bug somewhere else.

Reply:
> such a nil is a bug somewhere else.

I won't argue it, but this is consistent with the rest of our admission plugins and I'm disinclined to change them at this juncture.